### PR TITLE
Refactor: simplified Parallel_2d interface

### PR DESCRIPTION
**Pull request by jinzx10**
### Reminder
- [ ] Have you linked an issue with this pull request?
- [ ] Have you added adequate unit tests and/or case tests for your pull request?
- [ ] Have you noticed possible changes of behavior below or in the linked issue?
- [ ] Have you explained the changes of codes in core modules of ESolver, HSolver, ElecState, Hamilt, Operator or Psi? (ignore if not applicable)

### Unit Tests and/or Case Tests for my changes
- The unit test for BLACS functions are redesigned to be more informative.

### What's changed?
Parallel_2D, which is the base class of Parallel_Orbitals, has its user interface simplified in order to facilitate the use of 2d block-cyclic distrubtion.

### Any changes of core modules? (ignore if not applicable)
No.

### CODE REVIEW

File: source/module_basis/module_ao/parallel_2d.cpp [hunk 1]
```diff
@@ -1,209 +1,127 @@
...
+    this->blacs_ctxt = blacs_ctxt;
+    Cblacs_gridinfo(blacs_ctxt, &dim0, &dim1, &coord[0], &coord[1]);
+    _set_dist_info(mg, ng, nb);
+    return nrow == 0 || ncol == 0;
 }
 #endif
+
+void Parallel_2D::set_serial(const int mg, const int ng)
+{
+    nb = 1;
```
Comment (mohanchen) [inline source/module_basis/module_ao/parallel_2d.cpp:113]: Would it be better to assert(mg>0) and assert(ng>0)?

File: source/module_basis/module_ao/parallel_2d.h [hunk 2]
```diff
@@ -12,117 +13,112 @@
...
-    int* global2local_row_ = nullptr;
-    int* global2local_col_ = nullptr;
+    std::vector<int> global2local_row_;
+    std::vector<int> global2local_col_;
 
     /// map from local index to global index
-    std::vector<int> local2global_row_;				// Peize Lin change int* to vector 2022.08.03
+    std::vector<int> local2global_row_;
     std::vector<int> local2global_col_;
+    // Peize Lin change int* to vector 2022.08.03
```
Comment (mohanchen) [inline source/module_basis/module_ao/parallel_2d.h]: I think this line can be deleted.

File: source/module_basis/module_ao/parallel_2d.cpp (hunk 1 above)
Comment (jinzx10) [reply source/module_basis/module_ao/parallel_2d.cpp:113]: An assertion has been added in the latest commit.

File: source/module_basis/module_ao/parallel_2d.h (hunk 2 above)
Comment (jinzx10) [reply source/module_basis/module_ao/parallel_2d.h]: This line has been removed in the latest commit.

**Review by mohanchen — APPROVED** [review]
Great job!!!

**Pull request merged**